ouroboros-consensus-cardano/src/shelley/Ouroboros/Consensus/Shelley/Ledger/Query.hs

```diff
@@ -283,6 +286,14 @@ data instance BlockQuery (ShelleyBlock proto era) :: Type -> Type where
     => Set (SL.Credential 'SL.Staking (EraCrypto era))
     -> BlockQuery (ShelleyBlock proto era) (VoteDelegatees (EraCrypto era))
 
+  -- | Obtain a snapshot of big ledger peers. CLI can serialize these,
+  -- and if made available to the node by topology configuration,
+  -- the diffusion layer can use these peers when syncing up from scratch
+  -- or stale ledger state - especially useful for Genesis mode
+  GetPeerSnapshot
```

Comment:
In case you do not already have golden tests upstream: You could add an example query + result here:
https://github.com/IntersectMBO/ouroboros-consensus/blob/568897ddfab3de0932ec16a477df46d8bb41ac56/ouroboros-consensus-cardano/src/unstable-shelley-testlib/Test/Consensus/Shelley/Examples.hs#L106-L123
in order to be sure that the serialization format is not accidentally changed in the future. (also see https://github.com/IntersectMBO/ouroboros-consensus/issues/167)

Reply:
Done.